### PR TITLE
Syncing sync2/ to top of repo

### PR DESCRIPTION
**Pull request by danielrh**

### CODE REVIEW

File: sync2/cond_timed_wait.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	case <-done:
+		return true
+	case <-time.After(timeout):
+		if atomic.SwapInt32(decided, 1) == 0 {
+			// Timeout wins.
+			cond.L.Lock()
+			return false
+		} else {
+			// Wait wins
+			<-done
```
Comment (pattyshack) [inline sync2/cond_timed_wait.go]: rm.  just return true is sufficient.

File: sync2/cond_timed_wait.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	"time"
+)
+
+// Wait for a condition. If the timeout passes, returns 'false'.
+// If the condition is signalled, returns 'true'.
+//
+// The condition's mutex must already be held when calling this function.
+//
+// If reSignal is true, then the condition will be re-signaled on timeout.
+func CondTimedWait(cond *sync.Cond, timeout time.Duration, reSignal bool) bool {
```
Comment (pattyshack) [inline sync2/cond_timed_wait.go]: as discussed via email, we should remove the reSignal argument and unconditionally resignal on timeout

**Comment by pattyshack** [comment]
let's get rid of CondTimedWait (internally / externally) since no product code is using it and it's a bit sketchy.

**Comment by msolo-dropbox** [comment]
+1

**Pull request closed without merging**